backend/app/endpoints.py

```diff
@@ -123,12 +123,28 @@ async def get_portfolio(address: str, start=str):
     )
 
 
-@router.get("/backtest/spread/{address}/{start}/{token_to_divest_from}/{percentage}")
+@router.get(
+    "/backtest/spread/{address}/{start}/{token_to_divest_from}/{spread_token}/{percentage}"
+)
 async def backtest_spread(
-    address: str, start: str, token_to_divest_from: str, percentage: int
+    address: str,
+    start: str,
+    token_to_divest_from: str,
+    percentage: int = Path(0, ge=0, le=100),
```

Comment:
The first arg to `Path` (0 on this line or "USDC" the line below), seem to correspond to the `default` parameter to the `Path` constructor. There is no point in having a default value for these `Path` objects, as the URL needs to be well-defined anyways. So these arguments are misleading and don't provide any value, let's remove them.

Also, I would feel safer if we reorder the args to this function `backtest_spread` so that they correspond to the named placeholders of the format string given to `router.get` on line 127. Namely, `spread_token` then `percentage`. Not sure if it's needed or just a matter of preference, though.

Reply:
Unfortunately, `default` is a non-optional positional argument in FastAPI 0.77.1, it is not required for [>=0.78.0 however](https://fastapi.tiangolo.com/release-notes/?h=path%28#0780).

Despite this, I think we should keep this `Path` syntax as it provides better path validation in the API. For example, the API returns an Error 422 if `percentage` is out of bounds as well as a response body that clearly states the issue,
```
{"detail":[{"loc":["path","percentage"],"msg":"ensure this value is greater than or equal to 0","type":"value_error.number.not_ge","ctx":{"limit_value":0}}]}
```
which can be propagated to the frontend if desired, notifying the user of their error. Otherwise, the error would be an ambiguous Error 500 with "Internal Server Error" as the response body which can be any number of issues besides the `percentage` being out of bounds.

It also makes the code cleaner, as the validation does not have to be done within the function body.

Your second remark has been noted 👍.

Reply:
Indeed, using `Path` is great. So let's keep things this way until we upgrade FastAPI.